crates/witness-generator/src/eest_generator.rs

```diff
@@ -220,30 +222,137 @@ fn find_all_files_with_extension(path: &Path, extension: &str) -> Vec<PathBuf> {
 
-            assert_eq!(
-                !directory_path.exists(),
-                delete_eest_folder,
-                "Directory should {}exist after drop",
-                if delete_eest_folder { "" } else { "not " }
-            );
-        }
+        let single_fixture_path =
+            PathBuf::from_str("fixtures/blockchain_tests/zkevm/worst_compute/worst_jumps.json")?;
+        std::fs::create_dir_all(&target_path.join(single_fixture_path.parent().unwrap()))?;
+        std::fs::copy(
+            decompress_path
+                .join("zkevm-fixtures")
+                .join(&single_fixture_path),
+            target_path.join(&single_fixture_path),
+        )?;
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_custom_input_folder() -> Result<()> {
+        let target_dir = tempfile::tempdir()?;
+        let target_path = target_dir.path();
+        prepare_downgraded_eest_fixtures(&target_path)?;
+
+        let wg = ExecSpecTestBlocksAndWitnessBuilder::default()
+            .with_input_folder(target_path.to_path_buf())?
+            .build()?;
+
+        // The worst_jumps.json suite has two fixtures.
+        assert_eq!(
+            wg.generate().await?.len(),
+            2,
+            "Only two fixtures are expected for the keccak EEST fixture"
+        );
+
+        // Then the `input_folder` is used, the folder must not be deleted.
+        drop(wg);
+        assert!(
+            target_path.exists(),
+            "Directory should still exist after drop"
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_filters() -> Result<()> {
+        let target_dir = tempfile::tempdir()?;
+        let target_path = target_dir.path();
+        prepare_downgraded_eest_fixtures(&target_path)?;
+
+        let bw_with_include = ExecSpecTestBlocksAndWitnessBuilder::default()
+            .with_input_folder(target_path.to_path_buf())?
+            .with_includes(vec!["Prague".to_string()])
+            .build()?
+            .generate()
+            .await?;
+        assert_eq!(
+            bw_with_include.len(),
+            1,
+            "Only one fixture should match the include filter"
+        );
+        assert!(
+            bw_with_include[0].name.contains("Prague"),
+            "The fixture should contain 'Prague' in its name"
+        );
+
+        let bw_with_exclude = ExecSpecTestBlocksAndWitnessBuilder::default()
+            .with_input_folder(target_path.to_path_buf())?
+            .with_excludes(vec!["Prague".to_string()])
+            .build()?
+            .generate()
+            .await?;
+        assert_eq!(
+            bw_with_exclude.len(),
+            1,
+            "Only one fixture should match the exclude filter"
+        );
+        assert!(
+            !bw_with_exclude[0].name.contains("Prague"),
+            "The fixture should not contain 'Prague' in its name"
+        );
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    #[cfg(feature = "slow-tests")]
```

Comment:
This test is quite heavy since it runs witnesses for all fixtures. But as mentioned before today's only filling is for 36M which is quite heavy.

For now I'm leaving this under a `--features slow-tests` flag, and out of the CI since it's too heavy. As soon as they're downgraded for 1M baked into the release, that might be OK for GH machines and we can remove the feature. We can figure out then.

Reply:
Yep, another solution might be to run these using GitHub corn jobs once a day if that's useful